### PR TITLE
Feat/user data

### PR DESCRIPTION
**Pull request by bogdanbiv**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/meta/userdb/userdb.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+type UserMetadata struct {
+	// data for each user.
+	StarredRepos    []string
+	BookmarkedRepos []string
+}
+
+type UserMetadataEmptyStore struct{}
+
+//nolint:dupl
```
Comment (bogdanbiv) [inline pkg/meta/userdb/userdb.go:51]: Query description: get stars for user with ID / receive list of repo names

File: pkg/meta/errors/metadata.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package errors
+
+import "errors"
+
+var (
+	ErrUnableToCreateUserBucket       = errors.New("unable to create a user bucket for user")
```
Comment (andaaron) [inline pkg/meta/errors/metadata.go:6]: I think these should be moved in the errors package under the root of the repo.
@rchincha what do you think?

File: pkg/meta/userdb/userdb.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
+package userdb
```
Comment (andaaron) [inline pkg/meta/userdb/userdb.go:1]: I think this package should be under pkg/meta/userdb/boltdb-wrapper/userdb.go, instead of directly udner userdb.

File: pkg/storage/common.go [hunk 4]
```diff
@@ -9,7 +9,7 @@ import (
 
 	"github.com/docker/distribution/registry/storage/driver"
 	"github.com/gobwas/glob"
-	"github.com/notaryproject/notation-go"
+	notation "github.com/notaryproject/notation-go"
```
Comment (andaaron) [inline pkg/storage/common.go:12]: Why did this change? Also isn't this breaking anything?

File: pkg/storage/local/local.go [hunk 5]
```diff
@@ -46,6 +46,12 @@ const (
 	defaultSchemaVersion = 2
 )
 
+// BlobUpload models and upload request.
+type BlobUpload struct {
```
Comment (andaaron) [inline pkg/storage/local/local.go:50]: Is this change needed?

File: test/blackbox/metadata.bats [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                ]
+            }
+        }
+    },
+    "log": {
+        "level": "debug"
+    }
+}
+EOF
+    git -C ${BATS_FILE_TMPDIR} clone https://github.com/project-zot/helm-charts.git
```
Comment (andaaron) [inline test/blackbox/metadata.bats:56]: What is this used for?

File: test/blackbox/metadata.bats [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+@test "push image user policy" {
+    run skopeo --insecure-policy copy --dest-creds test:test --dest-tls-verify=false \
+        oci:${TEST_DATA_DIR}/golang:1.18 \
+        docker://127.0.0.1:8080/golang:1.18
+    [ "$status" -eq 0 ]
+}
+
+@test "User metadata starredRepos" {
+    run skopeo --insecure-policy copy --dest-tls-verify=false \
```
Comment (andaaron) [inline test/blackbox/metadata.bats:78]: Isn't this supposed to fail, because as anonymous you cannot push?

**Comment by andaaron** [comment]
Superseded by https://github.com/project-zot/zot/pull/1317
